eclair-core/src/main/scala/fr/acinq/eclair/wire/Onion.scala

```diff
@@ -168,6 +168,22 @@ object Onion {
 
   import OnionTlv._
 
+  /*
+   * We use the following architecture for onion payloads:
+   *
+   *                                                              PerHopPayload
+   *                                           _______________________/\__________________________
+   *                                          /                                                   \
+   *                                 RelayPayload                                              FinalPayload
+   *                     _______________/\_________________                                    ____/\______
+   *                    /                                  \                                  /            \
+   *           ChannelRelayPayload                          \                                /              \
+   *         ________/\______________                        \                              /                \
+   *        /                        \                        \                            /                  \
+   * RelayLegacyPayload     ChannelRelayTlvPayload     NodeRelayPayload          FinalLegacyPayload     FinalTlvPayload
```

Comment:
Shouldn't be rename `RelayLegacyPayload` -> `ChannelRelayLegacyPayload`?

Reply:
We could, but legacy already implies only channel relay. And since we're adding more dimensions to those type at some point we won't be able to mention all the dimensions inside each concrete type's name so we shouldn't strive for that :sweat_smile: